algorithms/ar-bsrh/javascript/BinarySearch.js

```diff
@@ -0,0 +1,34 @@
+function binarySearch(items, value) {
+	var firstIndex = 0,
+		lastIndex = items.length - 1,
+		middleIndex = Math.floor((lastIndex + firstIndex) / 2);
+
+	while (items[middleIndex] !== value && firstIndex < lastIndex) {
+		if (value < items[middleIndex]) {
+			lastIndex = middleIndex - 1;
+		}
+		else if (value > items[middleIndex]) {
+			firstIndex = middleIndex + 1;
+		}
+		middleIndex = Math.floor((lastIndex + firstIndex) / 2);
+	}
+
+	return items[middleIndex] !== value ? -1 : middleIndex;
+}
+
+function main() {
+	let items = [
+		1,
+		2,
+		3,
+		4,
+		5,
+		7,
+		8,
+		9
+	];
+	binary_Search(items, 1);
```

Comment:
Codacy found an issue: ['binary_Search' is not defined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6275682)